### ISSUE TITLE
FilterFunc in the informer is not working on the first run of reconciler

### ISSUE THREAD

**Issue by danyinggu** [knative/pkg#1764]
## Expected Behavior
The informer should always successfully filter objects based on filterFunc defined the informer

## Actual Behavior
In the first run of the reconcile, it lists all objects and objects are not filtered by filterFunc in the informer 

## Steps to Reproduce the Problem

1. Given a controller with an informer that has a filterFunc
2. Deploy the controller 
3. In the first run of the reconciler, it reconciles all the objects but not only the objects should pass the filterFunc

## Additional Info
We did some investigation, in the genreconciler: https://github.com/google/knative-gcp/blob/3707bd3ca3d010532b334201dc7da6c9bc382959/pkg/client/injection/reconciler/broker/v1beta1/broker/controller.go#L73-L81
When a controller starts, a new leader is promoted with a listAll operation and there is no informer resync.

**Comment by danyinggu** [comment]
@mattmoor @n3wscott @grantr

**Comment by danyinggu** [comment]
kind/bug

**Comment by mattmoor** [comment]
Yup, there's no filter passed to genreconciler currently, but you could plumb one through in the controller options.

**Comment by grantr** [comment]
@mattmoor can you point to an example of plumbing through? Looks like the code directly enqueues every object in the lister, skipping any informer FilterFunc.

**Comment by mattmoor** [comment]
By "plumbing through" I mean adding a field and calling it from that logic in genreconciler itself, there is no field today.
